pages/api/pluscode/[pluscode].js

```diff
@@ -0,0 +1,24 @@
+import app from '@/app';;
+import InternalError from '@/errors/InternalError';
+import getLocation from '@/services/pluscode/pluscode.js';
+
+const action = async (request, response) => {
+  try {
+    const plusCode = request.query.pluscode;
+    const data = await getLocation(plusCode);
+
+    response.status(200);
+    response.json(data);
+  } catch (error) {
+    
+    throw new InternalError({
+      status: 400,
+      message: `Plus Code inválido`,
+      name: 'PLUS_CODE_ERROR',
+      type: 'PLUS_CODE_ERROR',
+    });
+  }
+};
+
+
+export default app().get(action);
```

Comment:
Conhecendo a api do open street maps de experiências passadas, e sabendo que essa informação não deve mudar com muita frequência eu recomendaria usar um cache mais agressivo que o padrão do projeto, assim não corremos o risco de blocks.. .rs
```suggestion
export default app( { cache: 600000 } ).get(action);
```